prov/rxd/src/rxd_cq.c

```diff
@@ -307,7 +307,18 @@ void rxd_progress_tx_list(struct rxd_ep *ep, struct rxd_peer *peer)
 			}
 			continue;
 		}
-
+				
+		if (tx_entry->op == RXD_DATA_READ && !tx_entry->bytes_done) {
+			if (ep->peers[tx_entry->peer].unacked_cnt >=
+		    	    ep->peers[tx_entry->peer].tx_window) {
+				break;
+			} else {
+				tx_entry->start_seq = ep->peers[tx_entry->peer].tx_seq_no;
+				ep->peers[tx_entry->peer].tx_seq_no = tx_entry->start_seq +
+							      	      tx_entry->num_segs;
+			}
+		}
+		
 		if (!rxd_ep_post_data_pkts(ep, tx_entry))
```

Comment:
Per Sean's email. I think we should check this and return EAGAIN if we don't have enough memory. This should happen by default if we don't return any completions so I don't think there's too much you need to do to return this. But we do have to make sure that if that happens that we reverse the tx_seq_no increment that we just did so we don't skip these sequence numbers.